src/levels.cpp

```diff
@@ -1024,13 +1024,45 @@ bool levLoadData(char const *name, Sha256 const *hash, char *pSaveName, GAME_TYP
 			return false;
 		}
 	}
-
+	// this will trigger upgrades 
 	if (!stageThreeInitialise())
 	{
 		debug(LOG_ERROR, "Failed stageThreeInitialise()!");
 		return false;
 	}
 
+	// droids were actually loaded before (during stageThreeInitialise) upgrades.
+	// So to actually apply them, we'll iterate over all droids, and re-calculate, once again,
+	// their upgradable parts.
+	// Without that, both Campaign and Skirmish saves are broken.
+	for (int player=0; player < MAX_PLAYERS; player++)
+	{
+		// those are droids present on the map
```

Comment:
What if someone created a save while a transporter was on map but it hasn't landed yet?

Reply:
no idea. I ll try that

Reply:
Goddamit
I ll fix that

We definitely need a simple way to iterate over all droids, this is ridiculous